llvmlite/ir/types.py

```diff
@@ -193,7 +193,9 @@ def gep(self, i):
 
     @property
     def intrinsic_name(self):
-        return 'p%d%s' % (self.addrspace, self.pointee.intrinsic_name)
+        if ir_layer_typed_pointers_enabled:
```

Comment:
I think this is a fix for an error that was previously un-noticed / un-addressed - is that correct?

Reply:
Yep, it fixes an inconsistent behaviour in how pointer intrinsic names are printed.

Previously, even if we enabled IR layer typed pointers, the pointer's intrinsic name would still reflect its "true" underlying type. This is probably not the desired behaviour, so I've changed this to follow the same logic we have for the "_to_string" method.

I should have done this when we added IR layer typed pointers, but I think this got lost with the opaque pointer fixes (which in practice came _after_ IR layer typed pointers, but I had initially thought them to come before).